### PR TITLE
Port our Collection code, docs, tests to use ansible-galaxy cli

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #5335
https://pulp.plan.io/issues/5335

### CODE REVIEW

**Comment by kersommoura** [comment]
@daviddavis, those changes look good to me. I will check again once CI runs.

File: pulp_ansible/tests/functional/api/collection/test_consume_content.py [hunk 1]
```diff
@@ -55,12 +57,15 @@ def test_consume_content(self):
         )
         self.addCleanup(self.client.delete, distribution["_href"])
 
+        collection_dir = tempfile.mkdtemp()
```
Comment (kersommoura) [inline pulp_ansible/tests/functional/api/collection/test_consume_content.py]: We could add a clean up for this dir. Not a blocker.

Comment (daviddavis) [reply pulp_ansible/tests/functional/api/collection/test_consume_content.py]: I figured since it was a temp dir, we wouldn't have to worry about it?

Comment (daviddavis) [reply pulp_ansible/tests/functional/api/collection/test_consume_content.py]: Do we have to cleanup temp directories?

Comment (bmbouter) [reply pulp_ansible/tests/functional/api/collection/test_consume_content.py]: The temp facilities will handle this for us is my understanding.

**Pull request merged**